Q1.	The time required for servicing transmissions is normally distributed with µ = 45 minutes and σ = 8 minutes. The service manager plans to have work begin on the transmission of a customer’s car 10 minutes after the car is dropped off and the customer is told that the car will be ready within 1 hour from drop-off. What is the probability that the service manager cannot meet his commitment? 

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [3]:
#Let 'X' be the amount of time it takes to complete the repair on a customer's car.
#So the Probability of completing the car repair with 1 hr of dropping the car and starting the work after 10 mins of car drop (60 mins - 10 mins) is P(X<=50 mins)
#So, P(X>50) = 1-P(X<=50) i.e. cannot meet his commitment
# Find Z-Scores at X=50; Z = (X - µ) / σ 
# Given µ = 45  and σ = 8

In [4]:
Z=(50-45)/8
Z

0.625

In [7]:
# Find probability P(X>50) = 1-stats.norm.cdf(abs(z_score))
1-stats.norm.cdf(abs(0.625))

0.26598552904870054

### Q2.	The current age (in years) of 400 clerical employees at an insurance claims processing center is normally distributed with mean µ = 38 and Standard deviation σ = 6. For each statement below, please specify True/False. If false, briefly explain why.
A.	More employees at the processing center are older than 44 than between 38 and 44.

B.	A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.

In [8]:
# gIVEN µ = 38, σ = 6, N = 400
# using stats.norm.cdf(x, loc = mean, scale= std )

In [19]:
#A. More employees at the processing center are older than 44 than between 38 and 44.
# employees at the processing center are older than 44           p(x>44)
1- stats.norm.cdf(x=44,loc=38,scale=6)

0.15865525393145707

In [21]:
#employees at the processing center between 38 and 44.          p(38<X<44)
stats.norm.cdf(x=44,loc=38,scale=6)-stats.norm.cdf(x=38,loc=38,scale=6)

0.3413447460685429

THE ABOVE STATEMENT IS FALSE AS PER OUR CALCULATION THE PROBABILITY OF Employees at the processing center
between 38 and 44 IS MORE THAN employees at the processing center are older than 44 

In [23]:
#B. A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.
# gIVEN µ = 38, σ = 6 , N= 400, P(X<30)  ,  
# using stats.norm.cdf(x, loc = mean, scale= std )

stats.norm.cdf(x=30,loc=38,scale=6)

0.09121121972586788

In [25]:
#using N*P(x<30)

400*0.091

36.4

From above, statement B is True as no. of employees aged below 33 yrs attending training is 36

Q4. Let X ~ N(100, 20^2). Find two values, a and b, symmetric about the mean, such that the probability of the random variable taking a value between them is 0.99. 

In [27]:
# Given x=0.99 , loc=100, scale=20
# using stats.norm.interval(0.99,loc,scale)
stats.norm.interval(0.99, loc=100, scale=20)

(48.48341392902199, 151.516586070978)

A=48.48 , B= 151.52

Q5. Consider a company that has two different divisions. The annual profits from the two divisions are independent and have distributions Profit1 ~ N(5, 3^2) and Profit2 ~ N(7, 4^2) respectively. Both the profits are in $ Million. Answer the following questions about the total profit of the company in Rupees. Assume that $1 = Rs. 45
A.	Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.
B.	Specify the 5th percentile of profit (in Rupees) for the company
C.	Which of the two divisions has a larger probability of making a loss in a given year?

In [31]:
# Mean profits from two different divisions of a company = Mean1 + Mean2
Mean = 5+7
Mean

12

In [33]:
#given 1 dollar = Rs. 45
# will convert everything into dollar by multiplying them with 45
Mean_in_dollar = Mean*45
Mean_in_dollar


540

In [35]:
# Variance of profits from two different divisions of a company = SD^2 = SD1^2 + SD2^2

SD= np.sqrt((9)+(16))
SD

5.0

In [36]:
#given 1 dollar = Rs. 45
SD_in_dollar= SD*45
SD_in_dollar

225.0

In [38]:
# A. Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.
stats.norm.interval(0.95,540,225)

(99.00810347848784, 980.9918965215122)

Range is Rs (99.00810347848784, 980.9918965215122) in Millions

In [39]:
# B. Specify the 5th percentile of profit (in Rupees) for the company
# To compute 5th Percentile, we use the formula X=μ + Zσ; wherein from z table, 5 percentile = -1.645
X= 540+(-1.645)*(225)
X

169.875

5th percentile of profit (in Million Rupees) is 170.0

In [40]:
# C. Which of the two divisions has a larger probability of making a loss in a given year?

# Probability of Division 1 making a loss P(X<0)
stats.norm.cdf(0,5,3)

0.0477903522728147

In [41]:
# Probability of Division 2 making a loss P(X<0)
stats.norm.cdf(0,7,4)


0.040059156863817086

Probability of Division 1 making a loss in a given year is more than Division 2.